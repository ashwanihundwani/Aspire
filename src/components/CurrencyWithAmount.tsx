import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { CommonString, AppColors, layouting } from '../utils/constants'
import { avenirFont, FontStyle } from '../utils/utils'

const spacingFactor = layouting.spacingFactor

interface CurrencyWithAmountProps {
  amount: string,
  textColor?: string
}

const CurrencyWithAmount: React.FC<CurrencyWithAmountProps> = (props: CurrencyWithAmountProps) => {

  return (
    <View style={styles.balanceContent}>
      <View style={styles.balanceLabelContainer}>
        <Text style={styles.balanceLabel}>{CommonString.currencySign}</Text>
      </View>
      <Text style={[styles.balanceValue, props.textColor ? { color: props.textColor } : {}]}>{props.amount}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  balanceContainer: {
    marginTop: spacingFactor
  },
  balance: {
    fontSize: 14,
    color: AppColors.whiteColor,
    marginLeft: spacingFactor,
    fontFamily: avenirFont(FontStyle.Medium),
  },
  balanceContent: {
    flexDirection: "row",
    marginLeft: spacingFactor,
    marginTop: 10,
  },
  balanceLabelContainer: {
    marginTop: 5,
    borderRadius: 4,
    width: 40,
    height: 22,
    backgroundColor: AppColors.lightGreen,
  },
  balanceLabel: {
    color: AppColors.whiteColor,
    fontFamily: avenirFont(FontStyle.Bold),
    fontSize: 12,
    textAlign: "center",
    marginTop: 4
  },
  balanceValue: {
    color: AppColors.whiteColor,
    fontFamily: avenirFont(FontStyle.Bold),
    fontSize: 24,
    textAlign: "center",
    marginLeft: 10,
  }
})

export default React.memo<CurrencyWithAmountProps>(CurrencyWithAmount)