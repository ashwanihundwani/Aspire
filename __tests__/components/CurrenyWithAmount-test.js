/**
 * @format
 */

import 'react-native';
import React from 'react';
import CurrencyWithAmount from '../../src/components/CurrencyWithAmount';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
      .create(<CurrencyWithAmount amount={"3,000"}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});