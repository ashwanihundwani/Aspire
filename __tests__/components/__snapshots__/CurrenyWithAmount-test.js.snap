// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginLeft": 43.5,
      "marginTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#01D167",
        "borderRadius": 4,
        "height": 22,
        "marginTop": 5,
        "width": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "AvenirNext-Bold",
          "fontSize": 12,
          "marginTop": 4,
          "textAlign": "center",
        }
      }
    >
      S$
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "AvenirNext-Bold",
          "fontSize": 24,
          "marginLeft": 10,
          "textAlign": "center",
        },
        Object {},
      ]
    }
  >
    3,000
  </Text>
</View>
`;
